### PR TITLE
Introduce cgroups v2 memory controller 

### PR DESCRIPTION
**Pull request by dmah42**

### CODE REVIEW

File: api/v0/cells/cells.proto
Comment (future-highway) [inline api/v0/cells/cells.proto:91]: Nothing to do with this PR, but we need to make these double slashes and think about documenting a formatter. Is there an equivalent to rustfmt.toml for proto files?

File: api/v0/cells/cells.proto
Comment (future-highway) [inline api/v0/cells/cells.proto:167]: Can we make this `MemoryController` and `memory` to match the controller name as seen in the docs?

File: api/v0/cells/cells.proto
Comment (future-highway) [reply api/v0/cells/cells.proto:91]: To clarify, I don't expect this to be done in this PR. I was just noting my thoughts.

Comment (dmah42) [reply api/v0/cells/cells.proto:91]: i don't know about a rustfmt.toml, but things like prettier will format protos.

File: api/v0/cells/cells.proto
Comment (dmah42) [reply api/v0/cells/cells.proto]: done

File: api/v0/cells/cells.proto
Comment (future-highway) [inline api/v0/cells/cells.proto:243]: Low is used. High is not

**Review by future-highway — COMMENTED** [review]
I can't give a proper review but figured I'd leave a few comments. Generally looks good though.

File: auraed/src/cells/cell_service/cells/cgroups/cgroup.rs
Comment (dmah42) [reply auraed/src/cells/cell_service/cells/cgroups/cgroup.rs:135]: yeah i am surprised by this too. both cgroups-rs and libcgroups only implement partial memory control.  i figured it was useful for us to plumb this through so we can iterate towards more support without changing the API later.

File: auraed/src/cells/cell_service/cells/cgroups/cgroup.rs
Comment (dmah42) [reply auraed/src/cells/cell_service/cells/cgroups/cgroup.rs:93]: i had it separate and got into trouble trying to set cpu and memory on builder as copies started to creep in.  this felt clean too.

File: api/v0/cells/cells.proto
Comment (dmah42) [reply api/v0/cells/cells.proto:243]: some documentation may be out of date. :)

File: auraed/src/cells/cell_service/cells/cgroups/memory/mod.rs
Comment (future-highway) [inline auraed/src/cells/cell_service/cells/cgroups/memory/mod.rs:35]: Sorry, I wasn't clear. I think min is an Allocation as it is supposed to be guaranteed. Low is a protection and therefore not guaranteed.

Comment (dmah42) [reply auraed/src/cells/cell_service/cells/cgroups/memory/mod.rs:35]: in the docs, memory min is described as "Hard memory protection" and low as "Best-effort memory protection".  high is "Memory usage throttle limit" and max is "Memory usage hard limit".

so i think this is correct as written.

Comment (future-highway) [reply auraed/src/cells/cell_service/cells/cgroups/memory/mod.rs:35]: I agree with your reading, but still think the code is incorrect. Low and min shouldn't be the same type.

**Review by future-highway — APPROVED** [review]
Reviewing from my phone but everything looks good. I'll leave it to you dma if you want someone else to also review before merging.

**Review by future-highway — DISMISSED** [review]
Reviewing from my phone but everything looks good. I'll leave it to you dma if you want someone else to also review before merging.

**Review by krisnova — APPROVED** [review]

**Comment by krisnova** [comment]
Can we fix the conflicts and then merge this in please?

**Review by krisnova — APPROVED** [review]

**Pull request merged**
